e2e/ansible/playbooks/resiliency/test-node-failure/test-node-failure.yml

```diff
@@ -0,0 +1,223 @@
+
+       - name: Replace volume-claim name with test parameters
+         include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/regex_task.yml"
+         vars:
+           path: "{{ result_kube_home.stdout }}/percona.yaml"
+           regex1: "{{replace_item}}"
+           regex2: "{{replace_with}}"
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+
+       - name: Create namespace to deploy application
+         shell: source ~/.profile; kubectl create ns {{ namespace }}
+         args:
+           executable: /bin/bash
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+       - name: Create a configmap with the liveness sql script
+         shell: source ~/.profile; kubectl create configmap sqltest --from-file={{ percona_files.1 }} -n {{ namespace }}
+         args:
+           executable: /bin/bash
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+         register: result
+         failed_when: "'configmap' and 'created' not in result.stdout"
+
+       - include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_task.yml"
+         vars:
+           app_yml: "{{ percona_files.0 }}"
+           ns: "{{ namespace }}"
+
+       - include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+         vars:
+            ns: "{{ namespace }}"
+            app: percona
+
+       - name: Wait for 120s to ensure liveness check starts
+         wait_for:
+           timeout: 120
+
+       - name: Get PV name
+         shell: source ~/.profile; kubectl get pv -n {{ namespace }}| grep openebs-percona | awk '{print $1}'
+         args:
+           executable: /bin/bash
+         register: pv_name
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+
+       - name: Get the node name where replica is scheduled
+         shell: source ~/.profile; kubectl get po -n {{ namespace }} -o wide | grep "{{ pv_name.stdout }}" | grep rep | awk '{print $7}' | awk 'FNR == 1 {print}'
+         args:
+           executable: /bin/bash
+         register: node1
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+
+       - name: Bring down node
+         include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/vagrant-halt-up-task.yml"
+         vars:
+           vagrant_status: halt --force
+           nodename: "{{ node1.stdout }}"
+           vagrant_path: "{{ vagrant_path1 }}"
+
+
+       - name: check the node is down
+         shell: kubectl get nodes | grep {{ node1.stdout }} | awk '{print $2}'
+         args:
+           executable: /bin/bash
+         register: result
+         until: "'NotReady' in result.stdout"
+         delay: 60
+         retries: 5
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+       - name: Wait for 300s to check the pod is still running after killing the node
+         wait_for:
+           timeout: 300
+
+       - name: Confirm liveness checks on percona are successful &  pod is still in running state
```

Comment:
This check is being performed post fault cycle completion,  i.e., reverting the injected fault which should be good enough.